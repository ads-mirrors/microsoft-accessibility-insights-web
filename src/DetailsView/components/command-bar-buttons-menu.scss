// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

.command-bar-buttons-menu-button {
    font-size: 16px;
    padding: 9px !important;
    color: $primary-text !important;

    &:hover {
        background-color: $neutral-2 !important;
    }

    &:focus {
        outline: 1px solid;
    }
}

.menu-list {
    top: 19px !important;
}

.menu-item {
    padding-bottom: unset !important;

    &:hover {
        background: none !important;
    }
}

.menu-button {
    align-items: flex-start !important;
    font-weight: 400 !important;
    margin-left: -8px !important;

    &:hover {
        background: 'none !important';
        color: $insights-button-hover;

        & > span {
            & > svg {
                color: $insights-button-hover;
            }
        }
    }
}

.refresh-icon {
    color: $insights-button-hover !important;
}

.command-bar-buttons-submenu {
    min-width: fit-content;

    button {
        height: 36px;
    }
}

.command-bar-buttons-menu {
    height: 17px;
    margin-left: 9px !important;
    margin-top: 8px !important;
    margin-bottom: 16px !important;
}
