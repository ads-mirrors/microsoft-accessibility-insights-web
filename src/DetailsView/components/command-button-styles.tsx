// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { makeStyles, tokens } from '@fluentui/react-components';

export const useCommandButtonStyle = makeStyles({
    assessmentButton: {
        fontWeight: '400 !important',
        paddingLeft: '5px !important',
        textDecoration: 'none !important',

        '&:focus': {
            outline: '1px solid',
        },

        '&:hover': {
            background: 'none !important',
            color: tokens.colorNeutralForeground2BrandHover,

            '& > span': {
                '& >svg': {
                    color: tokens.colorNeutralForeground2BrandHover,
                },
            },
        },
    },
});
