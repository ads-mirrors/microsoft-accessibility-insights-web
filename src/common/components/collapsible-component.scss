// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

span.collapsible-title {
    padding-left: 0.5vw;
}

.collapsible {
    cursor: pointer;
    width: auto;
    border: none;
    outline: none;
    line-height: 0.25px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-right: 27px !important;
    padding-left: 1px !important;

    h2 {
        font-weight: $normal-title-font-weight;
        font-size: 17px;
    }
}

.collapsible-content {
    overflow: hidden;
    padding-left: calc(0.5vw + 13px); // 13px is Chevron width.
}
