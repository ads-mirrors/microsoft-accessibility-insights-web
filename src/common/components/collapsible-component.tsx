// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { css } from '@fluentui/utilities';
import styles from 'common/components/collapsible-component.scss';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import * as React from 'react';

export interface CollapsibleComponentProps {
    header: JSX.Element;
    content: JSX.Element;
    contentClassName?: string;
    containerClassName?: string;
}

interface CollapsibleComponentState {
    showContent: boolean;
}

export class CollapsibleComponent extends React.Component<
    CollapsibleComponentProps,
    CollapsibleComponentState
> {
    constructor(props: CollapsibleComponentProps) {
        super(props);
        this.state = { showContent: true };
    }

    private onClick = (): void => {
        this.setState(prevState => ({ showContent: !prevState.showContent }));
    };

    public render(): JSX.Element {
        const showContent = this.state.showContent;
        let content: JSX.Element | null = null;

        if (showContent) {
            content = (
                <div className={css(this.props.contentClassName, styles.collapsibleContent)}>
                    {this.props.content}
                </div>
            );
        }

        return (
            <div className={css(this.props.containerClassName)}>
                <Button
                    className={styles.collapsible}
                    appearance="transparent"
                    onClick={this.onClick}
                    aria-expanded={showContent}
                >
                    {showContent ? (
                        <FluentUIV9Icon iconName="ChevronDown24Regular" />
                    ) : (
                        <FluentUIV9Icon iconName="ChevronRight24Regular" />
                    )}
                    {this.props.header}
                </Button>
                {content}
            </div>
        );
    }
}
