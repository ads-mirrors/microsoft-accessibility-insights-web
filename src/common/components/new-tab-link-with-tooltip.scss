// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

:global(.high-contrast-theme) {
    .insights-tooltip {
        outline-style: solid;
        outline-width: 1px;
        outline-color: $always-white;
    }
}

.insights-tooltip-host {
    display: inline-block;
    min-width: 0;
}

.insights-link {
    display: flex;

    &:hover,
    &:active {
        text-decoration: underline !important;
    }
}

.insights-link-icon {
    display: flex;

    &:hover,
    &:active {
        text-decoration: none !important;
    }
}

@media screen and (forced-colors: active) {
    .insights-link:hover,
    .insights-link-icon:hover {
        forced-color-adjust: none;
        color: ButtonText !important;
    }
}
