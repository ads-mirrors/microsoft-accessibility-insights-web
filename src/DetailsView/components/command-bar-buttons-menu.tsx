// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    Menu,
    MenuButton,
    MenuList,
    MenuPopover,
    MenuProps,
    MenuTrigger,
    Tooltip,
} from '@fluentui/react-components';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import { ButtonRefFunction } from 'DetailsView/components/details-view-command-bar';
import { StartOverMenuItem } from 'DetailsView/components/start-over-component-factory';
import * as React from 'react';
import styles from './command-bar-buttons-menu.scss';

export type CommandBarButtonsMenuProps = {
    renderExportReportButton: () => JSX.Element | null;
    saveAssessmentButton?: JSX.Element | null;
    loadAssessmentButton?: JSX.Element | null;
    transferToAssessmentButton?: JSX.Element | null;
    getStartOverMenuItem: () => StartOverMenuItem;
    buttonRef?: ButtonRefFunction;
    hasSubMenu?: boolean;
};

export const CommandBarButtonsMenu = NamedFC<CommandBarButtonsMenuProps>(
    'CommandBarButtonsMenu',
    props => {
        const [open, setOpen] = React.useState(false);
        const onOpenChange: MenuProps['onOpenChange'] = (e, data) => setOpen(data.open);
        const exportButton = props.renderExportReportButton();

        const overflowItems: any[] = [];
        if (exportButton != null) {
            overflowItems.push({
                key: 'export report',
                children: <>{exportButton}</>,
            });
        }
        if (props.saveAssessmentButton && props.loadAssessmentButton) {
            overflowItems.push(
                {
                    key: 'save assessment',
                    children: <>{props.saveAssessmentButton}</>,
                },
                {
                    key: 'load assessment',
                    children: <>{props.loadAssessmentButton}</>,
                },
            );
        }

        if (props.transferToAssessmentButton) {
            overflowItems.push({
                key: 'transfer to assessment',
                children: <>{props.transferToAssessmentButton}</>,
            });
        }

        overflowItems.push({
            key: 'start over',
            ...props.getStartOverMenuItem(),
        });

        return (
            <>
                <Menu open={open} onOpenChange={onOpenChange}>
                    <Tooltip content="More actions" relationship="label">
                        <MenuTrigger disableButtonEnhancement>
                            <MenuButton
                                appearance="transparent"
                                aria-label="More actions"
                                icon={<FluentUIV9Icon iconName="MoreHorizontalRegular" />}
                                className={styles.commandBarButtonsMenuButton}
                                ref={() => props.buttonRef}
                            />
                        </MenuTrigger>
                    </Tooltip>
                    <MenuPopover
                        style={{
                            border: 'unset !important',
                            borderRadius: 'unset !important',
                        }}
                    >
                        <MenuList className={styles.menuList}>
                            {overflowItems.map((item, index) => (
                                <span key={index}>{item?.children}</span>
                            ))}
                        </MenuList>
                    </MenuPopover>
                </Menu>
            </>
        );
    },
);
