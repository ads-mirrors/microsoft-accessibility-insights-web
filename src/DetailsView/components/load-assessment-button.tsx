// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDataParser } from 'common/assessment-data-parser';
import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { TabStoreData } from 'common/types/store-data/tab-store-data';
import { UrlParser } from 'common/url-parser';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import { useCommandButtonStyle } from 'DetailsView/components/command-button-styles';
import { ButtonRefFunction } from 'DetailsView/components/details-view-command-bar';

import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
import * as React from 'react';

export type LoadAssessmentButtonDeps = {
    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
    assessmentDataParser: AssessmentDataParser;
    urlParser: UrlParser;
    loadAssessmentHelper: LoadAssessmentHelper;
};
export interface LoadAssessmentButtonProps {
    deps: LoadAssessmentButtonDeps;
    tabStoreData: TabStoreData;
    assessmentStoreData: AssessmentStoreData;
    handleLoadAssessmentButtonClick: () => void;
    isNarrowMode?: boolean;
    buttonRef?: ButtonRefFunction;
}

export const loadAssessmentButtonAutomationId = 'load-assessment-button';

export const LoadAssessmentButton = NamedFC<LoadAssessmentButtonProps>(
    'LoadAssessmentButton',
    props => {
        const loadButtonStyles = useCommandButtonStyle();
        return (
            <InsightsCommandButton
                data-automation-id={loadAssessmentButtonAutomationId}
                insightsCommandButtonIconProps={{
                    icon: <FluentUIV9Icon iconName="FolderOpenRegular" />,
                }}
                onClick={props.handleLoadAssessmentButtonClick}
                className={loadButtonStyles.assessmentButton}
                ref={props.buttonRef}
                {...props}
            >
                Load assessment
            </InsightsCommandButton>
        );
    },
);
