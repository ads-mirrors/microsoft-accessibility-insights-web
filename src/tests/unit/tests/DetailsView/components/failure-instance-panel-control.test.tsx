// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { TextField } from '@fluentui/react';
import { Link, Button } from '@fluentui/react-components';
import { act, fireEvent, render } from '@testing-library/react';
import { Assessments } from 'assessments/assessments';
import { FlaggedComponent } from 'common/components/flagged-component';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { CapturedInstanceActionType } from 'common/types/captured-instance-action-type';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { VisualizationType } from 'common/types/visualization-type';
import { ActionAndCancelButtonsComponent } from 'DetailsView/components/action-and-cancel-buttons-component';
import {
    FailureInstancePanelControl,
    FailureInstancePanelControlProps,
} from 'DetailsView/components/failure-instance-panel-control';
import { FailureInstancePanelDetails } from 'DetailsView/components/failure-instance-panel-details';
import { GenericPanel } from 'DetailsView/components/generic-panel';
import * as React from 'react';
import {
    expectMockedComponentPropsToMatchSnapshots,
    getMockComponentClassPropsForCall,
    mockReactComponents,
    useOriginalReactElements,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { IMock, Mock, Times } from 'typemoq';
jest.mock('@fluentui/react');
jest.mock('@fluentui/react-components');
jest.mock('common/components/flagged-component');
jest.mock('DetailsView/components/generic-panel');
jest.mock('DetailsView/components/action-and-cancel-buttons-component');
jest.mock('DetailsView/components/failure-instance-panel-details');
jest.mock('common/icons/fluentui-v9-icons');

describe('FailureInstancePanelControlTest', () => {
    mockReactComponents([
        Button,
        Link,
        GenericPanel,
        FlaggedComponent,
        TextField,
        ActionAndCancelButtonsComponent,
        FailureInstancePanelDetails,
        FluentUIV9Icon,
    ]);
    let addPathForValidationMock: IMock<(path) => void>;
    let addInstanceMock: IMock<(instanceData, test, step) => void>;
    let editInstanceMock: IMock<(instanceData, test, step, id) => void>;
    let clearPathSnippetDataMock: IMock<() => void>;

    beforeEach(() => {
        addInstanceMock = Mock.ofInstance(() => {});
        editInstanceMock = Mock.ofInstance(() => {});
        addPathForValidationMock = Mock.ofInstance(() => {});
        clearPathSnippetDataMock = Mock.ofInstance(() => {});
    });

    test('render FailureInstancePanelControl: create without instance', () => {
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);
        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    test('render FailureInstancePanelControl: partial original instance', () => {
        const props = {
            step: 'missingHeadings',
            test: VisualizationType.HeadingsAssessment,
            addFailureInstance: addInstanceMock.object,
            addPathForValidation: addPathForValidationMock.object,
            clearPathSnippetData: clearPathSnippetDataMock.object,
            actionType: CapturedInstanceActionType.CREATE,
            assessmentsProvider: Assessments,
            featureFlagStoreData: null,
            failureInstance: { failureDescription: 'original text' },
        };
        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        expect(renderResult.asFragment()).toMatchSnapshot();
        const flaggedProps = getMockComponentClassPropsForCall(FlaggedComponent);
        expect(flaggedProps.enableJSXElement.props.path).toBeUndefined();
    });

    test('render FailureInstancePanelControl: edit without instance', () => {
        const props = createPropsWithType(CapturedInstanceActionType.EDIT);
        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    test('closeFailureInstancePanel', () => {
        useOriginalReactElements('@fluentui/react', ['TextField']);
        const description = 'description';
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);
        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        const genericPanelProp = getMockComponentClassPropsForCall(GenericPanel);
        const textField = renderResult.getByRole('textbox') as HTMLInputElement;
        fireEvent.change(textField, { target: { value: description } });
        act(() => {
            genericPanelProp.onDismiss();
        });

        expect(genericPanelProp.isOpen).toBe(false);
        // This shouldn't be cleared because it stays briefly visible as the panel close animation happens
        expect(textField.value).toBe(description);

        clearPathSnippetDataMock.verify(handler => handler(), Times.exactly(2));
    });

    test('onFailureDescriptionChange', () => {
        useOriginalReactElements('@fluentui/react', [
            'TextField',
            'Panel',
            'ActionButton',
            'DefaultButton',
        ]);

        useOriginalReactElements('@fluentui/react-components', ['Link', 'Button']);
        useOriginalReactElements('DetailsView/components/generic-panel', ['GenericPanel']);
        useOriginalReactElements('common/components/flagged-component', ['FlaggedComponent']);
        useOriginalReactElements('DetailsView/components/action-and-cancel-buttons-component', [
            'ActionAndCancelButtonsComponent',
        ]);
        useOriginalReactElements('DetailsView/components/failure-instance-panel-details', [
            'FailureInstancePanelDetails',
        ]);
        const description = 'abc';
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);

        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);

        fireEvent.click(renderResult.getByText('Add a failure instance'));
        const textField = renderResult.getByRole('textbox') as HTMLInputElement;
        fireEvent.change(textField, { target: { value: description } });

        expect(textField.value).toBe(description);
    });

    test('onSelectorChange ', () => {
        const selector = 'some selector';
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);
        props.featureFlagStoreData = { manualInstanceDetails: true };
        props.failureInstance.path = '';
        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        fireEvent.click(renderResult.getByText('Add a failure instance'));
        const cssSelector = renderResult.getByLabelText('CSS Selector') as HTMLInputElement;
        fireEvent.change(cssSelector, { target: { value: selector } });
        expect(cssSelector.value).toBe(selector);
    });

    test('onValidateSelector ', () => {
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);

        const failureInstance = {
            failureDescription: 'new text',
            path: 'some selector',
            snippet: null,
        };

        props.failureInstance = failureInstance;
        props.featureFlagStoreData = { manualInstanceDetails: true };
        addPathForValidationMock
            .setup(handler => handler(failureInstance.path))
            .verifiable(Times.once());

        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        fireEvent.click(renderResult.getByText('Add a failure instance'));
        fireEvent.click(renderResult.getByText('Validate CSS selector'));

        addPathForValidationMock.verifyAll();
    });

    test('openFailureInstancePanel', () => {
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);
        const failureInstance = {
            failureDescription: 'new text',
            path: 'new path',
            snippet: 'new snippet',
        };
        props.failureInstance = failureInstance;
        props.featureFlagStoreData = { manualInstanceDetails: true };
        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        fireEvent.click(renderResult.getByText('Add a failure instance'));

        expect(renderResult.container.querySelector('.failureInstancePanel')).not.toBeNull;
        const failureDescription = renderResult.getByLabelText('Comment') as HTMLInputElement;
        expect(failureDescription.value).toEqual(props.failureInstance.failureDescription);
        const pathField = renderResult.getByLabelText('CSS Selector') as HTMLInputElement;
        expect(pathField.value).toEqual(props.failureInstance.path);
        const snippetText = renderResult.getByText(props.failureInstance.snippet);
        expect(snippetText).not.toBeNull();
    });

    test('onSaveEditedFailureInstance', () => {
        const description = 'text';
        const props = createPropsWithType(CapturedInstanceActionType.EDIT);
        props.instanceId = '1';
        props.editFailureInstance = editInstanceMock.object;

        const instanceData = {
            failureDescription: description,
            path: null,
            snippet: null,
        };

        editInstanceMock
            .setup(handler => handler(instanceData, props.test, props.step, props.instanceId))
            .verifiable(Times.once());

        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        fireEvent.click(renderResult.getByRole('button'));
        const textField = renderResult.getByRole('textbox') as HTMLInputElement;
        fireEvent.change(textField, { target: { value: description } });
        fireEvent.click(renderResult.getByText('Save'));
        expect(renderResult.container.querySelector('.failureInstancePanel')).not.toBeNull;
        editInstanceMock.verifyAll();
        clearPathSnippetDataMock.verify(handler => handler(), Times.exactly(2));
    });

    test('onAddFailureInstance', () => {
        const description = 'text';
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);

        const instanceData = {
            failureDescription: description,
            path: null,
            snippet: null,
        };

        addInstanceMock
            .setup(handler => handler(instanceData, props.test, props.step))
            .verifiable(Times.once());

        const renderResult = render(<FailureInstancePanelControl {...props} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        fireEvent.click(renderResult.getByText('Add a failure instance'));
        const textField = renderResult.getByRole('textbox') as HTMLInputElement;
        fireEvent.change(textField, { target: { value: description } });
        fireEvent.click(renderResult.getByText('Add failed instance'));
        expect(renderResult.container.querySelector('.failureInstancePanel')).not.toBeNull;
        addInstanceMock.verifyAll();
        clearPathSnippetDataMock.verify(handler => handler(), Times.exactly(2));
    });

    test('componentDidMount clears store', () => {
        const props = createPropsWithType(CapturedInstanceActionType.CREATE);
        const failureInstance = {
            failureDescription: null,
            path: 'inputed path',
            snippet: 'snippet for path',
        };
        props.failureInstance = failureInstance;

        const component = new FailureInstancePanelControl(props);
        component.componentDidMount();

        clearPathSnippetDataMock.verify(handler => handler(), Times.exactly(1));
    });

    test('componentDidUpdate reassigns state', () => {
        const prevProps = createPropsWithType(CapturedInstanceActionType.CREATE);
        const newProps = createPropsWithType(CapturedInstanceActionType.CREATE);
        const newFailureInstance = {
            failureDescription: null,
            path: 'inputed path',
            snippet: 'snippet for path',
        };
        prevProps.failureInstance.path = '';
        newProps.failureInstance = newFailureInstance;
        newProps.featureFlagStoreData = { manualInstanceDetails: true };
        prevProps.featureFlagStoreData = { manualInstanceDetails: true };
        const renderResult = render(<FailureInstancePanelControl {...prevProps} />);
        expectMockedComponentPropsToMatchSnapshots([Button, FlaggedComponent]);
        fireEvent.click(renderResult.getByText('Add a failure instance'));
        const pathField = renderResult.getByLabelText('CSS Selector') as HTMLInputElement;
        expect(pathField.value).toEqual(prevProps.failureInstance.path);
        const emptySnippetText = renderResult.getByText(
            'Code snippet will auto-populate based on the CSS selector input.',
        );
        expect(emptySnippetText).not.toBeNull();
        renderResult.rerender(<FailureInstancePanelControl {...newProps} />);
        expect(pathField.value).toEqual(newProps.failureInstance.path);
        const filledSnippetText = renderResult.getByText(newProps.failureInstance.snippet);
        expect(filledSnippetText).not.toBeNull();
    });

    function createPropsWithType(
        actionType: CapturedInstanceActionType,
    ): FailureInstancePanelControlProps {
        const featureData = {} as FeatureFlagStoreData;
        const emptyFailureInstance = {
            failureDescription: null,
            path: null,
            snippet: null,
        };

        return {
            step: 'missingHeadings',
            test: VisualizationType.HeadingsAssessment,
            addFailureInstance: addInstanceMock.object,
            addPathForValidation: addPathForValidationMock.object,
            clearPathSnippetData: clearPathSnippetDataMock.object,
            actionType: actionType,
            assessmentsProvider: Assessments,
            featureFlagStoreData: featureData,
            failureInstance: emptyFailureInstance,
        };
    }
});
