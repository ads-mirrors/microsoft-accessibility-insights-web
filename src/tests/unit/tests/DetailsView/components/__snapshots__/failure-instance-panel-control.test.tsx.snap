// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FailureInstancePanelControlTest closeFailureInstancePanel: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest closeFailureInstancePanel: FlaggedComponent props 1`] = `
{
  "enableJSXElement": <FailureInstancePanelDetails
    onSelectorChange={[Function]}
    onValidateSelector={[Function]}
    path={null}
    snippet={null}
  />,
  "featureFlag": "manualInstanceDetails",
  "featureFlagStoreData": {},
}
`;

exports[`FailureInstancePanelControlTest componentDidUpdate reassigns state: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest componentDidUpdate reassigns state: FlaggedComponent props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest onAddFailureInstance: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest onAddFailureInstance: FlaggedComponent props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest onFailureDescriptionChange: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest onFailureDescriptionChange: FlaggedComponent props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest onSaveEditedFailureInstance: Button props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest onSaveEditedFailureInstance: FlaggedComponent props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest onSelectorChange : Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest onSelectorChange : FlaggedComponent props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest onValidateSelector : Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest onValidateSelector : FlaggedComponent props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest openFailureInstancePanel: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest openFailureInstancePanel: FlaggedComponent props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: create without instance 1`] = `
<DocumentFragment>
  <mock-button
    appearance="transparent"
    aria-description="Open add a failure instance panel"
    aria-label="Add a failure instance"
    icon="[object Object]"
  >
    Add a failure instance
  </mock-button>
  <mock-genericpanel
    classname="failureInstancePanel"
    closebuttonarialabel="Close failure instance panel"
    hasclosebutton="true"
    headertext="Add a failure instance"
    isopen="false"
  >
    <mock-flaggedcomponent
      enablejsxelement="[object Object]"
      featureflag="manualInstanceDetails"
      featureflagstoredata="[object Object]"
    />
    <mock-styledtextfieldbase
      classname="observedFailureTextfield"
      label="Comment"
      multiline="true"
      placeholder="Comment"
      resizable="false"
      rows="8"
      value=""
    />
    <div>
      <mock-actionandcancelbuttonscomponent
        ishidden="false"
        primarybuttondisabled="true"
        primarybuttontext="Add failed instance"
      />
    </div>
  </mock-genericpanel>
</DocumentFragment>
`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: create without instance: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: create without instance: FlaggedComponent props 1`] = `
{
  "enableJSXElement": <FailureInstancePanelDetails
    onSelectorChange={[Function]}
    onValidateSelector={[Function]}
    path={null}
    snippet={null}
  />,
  "featureFlag": "manualInstanceDetails",
  "featureFlagStoreData": {},
}
`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: edit without instance 1`] = `
<DocumentFragment>
  <mock-link
    classname="editButton"
  />
  <mock-genericpanel
    classname="failureInstancePanel"
    closebuttonarialabel="Close failure instance panel"
    hasclosebutton="true"
    headertext="Edit failure instance"
    isopen="false"
  >
    <mock-flaggedcomponent
      enablejsxelement="[object Object]"
      featureflag="manualInstanceDetails"
      featureflagstoredata="[object Object]"
    />
    <mock-styledtextfieldbase
      classname="observedFailureTextfield"
      label="Comment"
      multiline="true"
      placeholder="Comment"
      resizable="false"
      rows="8"
      value=""
    />
    <div>
      <mock-actionandcancelbuttonscomponent
        ishidden="false"
        primarybuttondisabled="true"
        primarybuttontext="Save"
      />
    </div>
  </mock-genericpanel>
</DocumentFragment>
`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: edit without instance: Button props 1`] = `undefined`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: edit without instance: FlaggedComponent props 1`] = `
{
  "enableJSXElement": <FailureInstancePanelDetails
    onSelectorChange={[Function]}
    onValidateSelector={[Function]}
    path={null}
    snippet={null}
  />,
  "featureFlag": "manualInstanceDetails",
  "featureFlagStoreData": {},
}
`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: partial original instance 1`] = `
<DocumentFragment>
  <mock-button
    appearance="transparent"
    aria-description="Open add a failure instance panel"
    aria-label="Add a failure instance"
    icon="[object Object]"
  >
    Add a failure instance
  </mock-button>
  <mock-genericpanel
    classname="failureInstancePanel"
    closebuttonarialabel="Close failure instance panel"
    hasclosebutton="true"
    headertext="Add a failure instance"
    isopen="false"
  >
    <mock-flaggedcomponent
      enablejsxelement="[object Object]"
      featureflag="manualInstanceDetails"
    />
    <mock-styledtextfieldbase
      classname="observedFailureTextfield"
      label="Comment"
      multiline="true"
      placeholder="Comment"
      resizable="false"
      rows="8"
      value="original text"
    />
    <div>
      <mock-actionandcancelbuttonscomponent
        ishidden="false"
        primarybuttondisabled="false"
        primarybuttontext="Add failed instance"
      />
    </div>
  </mock-genericpanel>
</DocumentFragment>
`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: partial original instance: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-description": "Open add a failure instance panel",
  "aria-label": "Add a failure instance",
  "children": "Add a failure instance",
  "icon": <FluentUIV9Icon
    iconName="AddRegular"
  />,
  "onClick": [Function],
}
`;

exports[`FailureInstancePanelControlTest render FailureInstancePanelControl: partial original instance: FlaggedComponent props 1`] = `
{
  "enableJSXElement": <FailureInstancePanelDetails
    onSelectorChange={[Function]}
    onValidateSelector={[Function]}
  />,
  "featureFlag": "manualInstanceDetails",
  "featureFlagStoreData": null,
}
`;
